icu4c/source/common/localebuilder.cpp

```diff
@@ -167,14 +166,17 @@ _copyExtensions(const Locale& from, Locale* to, bool validate, UErrorCode& error
         CharString value;
         CharStringByteSink sink(&value);
         from.getKeywordValue(key, sink, errorCode);
+        if (U_FAILURE(errorCode)) { return; }
```

Comment:
Thanks for adding this failure check.
Just to explain, if getKeywordValue() fails, then the value is not usable, and we should not use value.data(), so we have to check here.